docs/contributor/compiler_walkthrough.md

```diff
@@ -127,6 +135,10 @@ The code generation (or codegen) step is where we generate the actual WebAssembl
 
 If you're curious about the wasm spec in general, you can check it out [here](https://webassembly.github.io/spec/core/index.html).
 
+## Linking
+
+In order to allow for users to split their code across multiple files we generate a mostly valid wasm output for each file. After this we merge all of the wasm files together into a single source file mapping the wasm imports and functions. This takes place in [linking/link.re](https://github.com/grain-lang/grain/blob/main/compiler/src/linking/link.re)
```

Comment:
@ospencer I don't know if you want to / have the time to expand on this. I think while my description is correct it might be a little to broad. I can also workshop this a little if wanted.

Reply:
```suggestion
Each Grain source file is compiled to a Grain-specific wasm file. To create the final program, we merge all of the files together in a step known as linking. This takes place in [linking/link.re](https://github.com/grain-lang/grain/blob/main/compiler/src/linking/link.re)
```